### PR TITLE
Email notification if failure occurs and log fixes

### PR DESCRIPTION
**Pull request by ives1227**
**Sends an email notification if an exception is thrown.  Also sets up time based file rollovers*
* * *

**GitHub Issue**: https://jira.huit.harvard.edu/browse/LTSEPADD-104

Testing:
Follow testing instructions here: https://github.com/harvard-lts/drs-import-management-service#test-environment-setup
I will send the .test.env's to the tester 

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? no
- integration tests? yes

### CODE REVIEW

File: test/functional/test_flask_app.py [hunk 1]
```diff
@@ -67,13 +68,15 @@ def setUp(self) -> None:
         self.dataverse_private_key_path = os.getenv('DATAVERSE_JWT_PRIVATE_KEY_FILE_PATH')
         self.epadd_private_key_path = os.getenv('EPADD_JWT_PRIVATE_KEY_FILE_PATH')
 
+    @unittest.skip('Broken- needs investigation')
```
Comment (awoods) [inline test/functional/test_flask_app.py:71]: Please include a link to a (newly created?) Jira ticket in this comment.

File: test/integration/test.env.example [hunk 2]
```diff
@@ -40,3 +40,11 @@ MQ_PROCESS_PASSWORD=admin
...
 MQ_PROCESS_QUEUE_PROCESS_READY=/queue/test-integration-dims-data-ready
 MQ_PROCESS_QUEUE_PROCESS_STATUS=/queue/test-integration-drs-ingest-status
+
+#Notify queuing
+MQ_HOST=b-cfb7f4c2-9140-4b5b-a551-1ccaf8c61d4e-1.mq.us-east-1.amazonaws.com
+MQ_PORT=61614
+MQ_USER=ims
+MQ_PASSWORD=XXX
+QUEUE_NAME=/queue/dais-notify
+DEFAULT_EMAIL_RECIPIENT=dts@hu.onmicrosoft.comå
```
Comment (awoods) [inline test/integration/test.env.example:50]: There appears to be an extraneous `å` at the end of the default email recipient address.

**Review by awoods — CHANGES_REQUESTED** [review]
Looks good. Some minor comments.

Unit and integration tests work, but functional tests return:
```
=========================== short test summary info ============================
FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_depositing_application_epadd_happy_path - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_expired_authorization_token - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_invalid_jwt_issuer - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_invalid_jwt_kid - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_invalid_request_body - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_jwt_not_corresponding_to_the_request_body - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
============= 6 failed, 1 passed, 2 skipped, 18 warnings in 15.82s =============
```

File: test/functional/test_flask_app.py (hunk 1 above)
Comment (ives1227) [reply test/functional/test_flask_app.py:71]: https://jira.huit.harvard.edu/browse/LTSEPADD-116

**Comment by ives1227** [comment]
> Looks good. Some minor comments.
> 
> Unit and integration tests work, but functional tests return:
> 
> ```
> =========================== short test summary info ============================
> FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_depositing_application_epadd_happy_path - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
> FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_expired_authorization_token - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
> FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_invalid_jwt_issuer - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
> FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_invalid_jwt_kid - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
> FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_invalid_request_body - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
> FAILED test/functional/test_flask_app.py::TestFlaskApp::test_ingest_endpoint_jwt_not_corresponding_to_the_request_body - FileNotFoundError: [Errno 2] No such file or directory: '/home/dimsuser/tes...
> ============= 6 failed, 1 passed, 2 skipped, 18 warnings in 15.82s =============
> ```

@awoods I think there was a step missing in the README.  I've updated it: https://github.com/harvard-lts/drs-import-management-service/blob/LTSEPADD-104/README.md#create-the-necessary-functional-testing-keys

Please let me know if that fixes the problem.

**Review by awoods — APPROVED** [review]
All tests now pass.

**Pull request merged**